### ISSUE TITLE
Service network_mode: "service: don't work in podman-compose version: 1.0.3 but in podman-compose version: 1.0.4

### ISSUE THREAD

**Issue by yodatak** [containers/podman-compose#593]
It work with `pip3 install https://github.com/containers/podman-compose/archive/devel.tar.gz
`
```yaml
version: '3'
services:
  web:
    image: busybox
    command: ["httpd", "-f", "-p", "8080", "-h", "/etc"]
  web2:
    image: busybox
    command: ["httpd", "-f", "-p", "8000", "-h", "/etc"]
    network_mode: "service:web"
```

**Expected behavior**

**Actual behavior**
What is the behavior you actually got and that should not happen.

```bash
podman-compose up -d
['podman', '--version', '']
using podman version: 4.2.1
** excluding:  set()
['podman', 'network', 'exists', 'testdockert_default']
podman run --name=testdockert_web_1 -d --label io.podman.compose.config-hash=123 --label io.podman.compose.project=testdockert --label io.podman.compose.version=0.0.1 --label com.docker.compose.project=testdockert --label com.docker.compose.project.working_dir=/opt/testdockert --label com.docker.compose.project.config_files=compose.yml --label com.docker.compose.container-number=1 --label com.docker.compose.service=web --net testdockert_default --network-alias web busybox httpd -f -p 8080 -h /etc
de2d219c3bbcaf836b91bf34fffa16d98f6ae8c26450942c54878d5e5d9839bb
exit code: 0
['podman', 'network', 'exists', 'testdockert_default']
podman run --name=testdockert_web2_1 -d --label io.podman.compose.config-hash=123 --label io.podman.compose.project=testdockert --label io.podman.compose.version=0.0.1 --label com.docker.compose.project=testdockert --label com.docker.compose.project.working_dir=/opt/testdockert --label com.docker.compose.project.config_files=compose.yml --label com.docker.compose.container-number=1 --label com.docker.compose.service=web2 --network service:web --net testdockert_default --network-alias web2 busybox httpd -f -p 8000 -h /etc
Error: invalid option for network service: unknown bridge network option: web
exit code: 125
podman start testdockert_web2_1
Error: no container with name or ID "testdockert_web2_1" found: no such container
exit code: 125
```

**Output**

```
podman-compose version
['podman', '--version', '']
using podman version: 4.2.1
podman-composer version  1.0.3
podman --version 
podman version 4.2.1
exit code: 0

podman info                  
host:
  arch: amd64
  buildahVersion: 1.27.0
  cgroupControllers:
  - cpuset
  - cpu
  - io
  - memory
  - hugetlb
  - pids
  - misc
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.1.4-2.fc37.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.4, commit: '
  cpuUtilization:
    idlePercent: 98.65
    systemPercent: 0.96
    userPercent: 0.39
  cpus: 8
  distribution:
    distribution: fedora
    variant: server
    version: "37"
  eventLogger: journald
  hostname: server.yodatak
  idMappings:
    gidmap: null
    uidmap: null
  kernel: 5.19.16-301.fc37.x86_64
  linkmode: dynamic
  logDriver: journald
  memFree: 22386204672
  memTotal: 33504157696
  networkBackend: netavark
  ociRuntime:
    name: crun
    package: crun-1.6-2.fc37.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.6
      commit: 18cf2efbb8feb2b2f20e316520e0fd0b6c41ef4d
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +LIBKRUN +YAJL
  os: linux
  remoteSocket:
    exists: true
    path: /run/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: false
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.2.0-8.fc37.x86_64
    version: |-
      slirp4netns version 1.2.0
      commit: 656041d45cfca7a4176f6b7eed9e4fe6c11e8383
      libslirp: 4.7.0
      SLIRP_CONFIG_VERSION_MAX: 4
      libseccomp: 2.5.3
  swapFree: 6354894848
  swapTotal: 8589930496
  uptime: 269h 12m 2.00s (Approximately 11.21 days)
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  volume:
  - local
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - docker.io
  - quay.io
store:
  configFile: /usr/share/containers/storage.conf
  containerStore:
    number: 3
    paused: 0
    running: 2
    stopped: 1
  graphDriverName: overlay
  graphOptions:
    overlay.mountopt: nodev,metacopy=on
  graphRoot: /var/lib/containers/storage
  graphRootAllocated: 248215093248
  graphRootUsed: 193058258944
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "true"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 2
  runRoot: /run/containers/storage
  volumePath: /var/lib/containers/storage/volumes
version:
  APIVersion: 4.2.1
  Built: 1662580673
  BuiltTime: Wed Sep  7 21:57:53 2022
  GitCommit: ""
  GoVersion: go1.19
  Os: linux
  OsArch: linux/amd64
  Version: 4.2.1


**Environment:**
 - OS:  Fedora 37

**Comment by yodatak** [comment]
Maybe a duplicate of ? https://github.com/containers/podman-compose/issues/379
But i see no test of this feature

**Comment by benyaminl** [comment]
If it's fixed on 1.0.4, then use the newer version, because the 1.0.4 is still in devel.. so... it won't be released soon...

**Comment by muayyad-alsadi** [comment]
I've just release 1.0.6 keeping behavior of no pod by default just like last stable release which 1.0.3

for 1.1.x --in-pod=1 will be the default.
